.github/workflows/build.yml

```diff
@@ -0,0 +1,79 @@
+name: Build Anmol
+
+on:
+  push:
+    branches: [ master, ibtida ]
+  workflow_dispatch:
+
+jobs:
+  build-anmol:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set testnet env
+        if: endsWith(github.ref, '/ibtida')
+        run: |
+          echo "ENVIRONMENT=testnet" >> $GITHUB_ENV
+
+      - name: Set production env
+        if: endsWith(github.ref, '/master')
+        run: |
+          echo "ENVIRONMENT=production" >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: |
+            /tmp/.buildx-cache-base
+            /tmp/.buildx-cache-prod
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER_NAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push base image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile.build
+          push: true
+          tags: anmolnetwork/anmol-node-build:latest-${{ env.TAG}}
```

Comment:
Re: versioning here. It seems you need to pass in a value to `build-args`, but I was not able to get this working with a second arg for the version. If we feel we absolutely need this right now, I can put some more time into getting it right, but otherwise it might be better to look into later.